libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/types.ts

```diff
@@ -0,0 +1,168 @@
+// (C) 2021 GoodData Corporation
+import { ObjRef } from "@gooddata/sdk-model";
+import isEmpty from "lodash/isEmpty";
+
+// Grantee types
+
+/**
+ * @internal
+ */
+export type GranteeItem = IGranteeUser | IGranteeGroup | IGranteeGroupAll;
+
+/**
+ * @internal
+ */
+export type GranteeType = "user" | "group" | "groupAll";
+
+/**
+ * @internal
+ */
+export interface IGranteeBase {
+    granteeType: GranteeType;
+    id: ObjRef;
+}
+
+/**
+ * @internal
+ */
+export interface IGranteeUser extends IGranteeBase {
+    granteeType: "user";
+    granteeName: string;
+    granteeEmail: string;
+    isOwner: boolean;
+    isCurrentUser: boolean;
+}
+
+/**
+ * @internal
+ */
+export const isGranteeUser = (obj: unknown): obj is IGranteeUser => {
+    return !isEmpty(obj) && (obj as IGranteeUser).granteeType === "user";
+};
+
+/**
+ * @internal
+ */
+export interface IGranteeGroup extends IGranteeBase {
+    granteeType: "group";
+    groupName: string;
+    granteeCount?: number;
```

Comment:
I think it should be possibly `memberCount` or `numberOfMembers` but this is minor. They are technically grantees so it's up to you.

Reply:
done